### PR TITLE
Used liked saved shows to make recommendation

### PR DESCRIPTION
**Pull request by alexisechano**
*Note: a lot of the code (especially in RecommendationClient) was pushed yesterday and this morning when I was setting up the Trakt API calls + making adjustments to UI

- There are now two ways to generate recommendations in FeedFragment: using liked, rated shows and the user's favorite genre
- Trending shows can still be viewed from the regular Feed
- User object now has list to track saved shows (couldn't be done before due to Parse's auth issues with subclassing)
- Added some UI edits like changing ratings to checkboxes to have a visual aid that something is rated and a placeholder for the profile image
- There is a walkthrough gif to show the UI edits and more but it is too big to attach here (will put in Workplace)

Screen Shot 2021-07-16 at 12 52 41 PM
Screen Shot 2021-07-16 at 12 52 58 PM

### CODE REVIEW

File: app/src/main/java/com/example/next_show/data/RecommendationClient.java [hunk 1]
```diff
@@ -29,20 +30,26 @@
...
-    public void fetchRelatedShows(Shows traktShows, Show compShow, ShowAdapter adapter) {
-        // use given show ID to generate related, recommended shows
-        String searchID = compShow.getId();
+    public void fetchRelatedShows(Shows traktShows, ShowAdapter adapter) {
+        // log statement
+        Log.i(TAG, "Fetching related shows from Trakt!");
+
+        // use given show ID to generate related, recommended shows -> randomize it!
+        List<String> savedShows = currentUser.getLikedSavedShows();
+        String searchID = savedShows.get(new Random().nextInt(savedShows.size()));;
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/data/RecommendationClient.java:52]: What's the usage of the random int here for?

File: app/src/main/java/com/example/next_show/data/RecommendationClient.java [hunk 2]
```diff
@@ -136,6 +142,7 @@ public void onResponse(Call<List<com.uwetrottmann.trakt5.entities.Show>> call, R
                         }
 
                     } else {
+                        Toast.makeText(context, "No shows available right now :(", Toast.LENGTH_LONG).show();
                         if (response.code() == UNAUTHORIZED_REQUEST) {
                             // authorization required, supply a valid OAuth access token
                             Log.e(TAG, "Access token required");
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/data/RecommendationClient.java:148]: I wuld move all of this and blow into a switch

File: app/src/main/java/com/example/next_show/data/RecommendationClient.java [hunk 3]
```diff
@@ -105,7 +111,7 @@ public void onFailure(Call<List<com.uwetrottmann.trakt5.entities.Show>> call, Th
     }
 
     // get it by genre from recommended shows call
-    public void fetchRecommendedShows(Shows traktShows, ShowAdapter adapter, User user) {
+    public void fetchRecommendedShows(Shows traktShows, ShowAdapter adapter) {
         // from trakt wrapper -> call to get related shows and send to adapter
         try {
             // enqueue to do asynchronous call and execute to do it synchronously
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/data/RecommendationClient.java:117]: This if else below is getting pretty long.  I would pull it out into two separate methods to improve readability

File: app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java [hunk 4]
```diff
@@ -108,7 +108,7 @@ private void setUpView(View currView) {
             btnSaveShow.setVisibility(View.GONE);
 
             // check if user already rated it, show if user liked it or not
-            if (currShow.getUserLiked() != null) {
+            if (currShow.getUserLiked().equals("liked") || currShow.getUserLiked().equals("disliked")) {
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java:111]: Do you not need to check `null` anymore?

File: app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java [hunk 5]
```diff
@@ -128,25 +128,28 @@ public void onClick(View v) {
                 currShow.setUserLiked("liked");
                 Toast.makeText(getActivity(), "You liked this show!", Toast.LENGTH_SHORT).show();
                 if(alreadySaved){
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java:130]: This if block is not empty.  Should the TODO still be there?

File: app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java [hunk 6]
```diff
@@ -164,7 +167,10 @@ public void done(ParseException e) {
 
                         // add show ID to User list
                         User currentUser = (User) ParseUser.getCurrentUser();
-                        currentUser.addToSavedShows(currShow.getId());
+
+                        if(currShow.getUserLiked().equals("liked")) {
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java:171]: Here and throughout -- code style there should be a space:
```
if (currShow....) {

}
```

File: app/src/main/java/com/example/next_show/models/Show.java [hunk 7]
```diff
@@ -201,6 +201,12 @@ public void setParseFields(ParseUser currUser) {
         setParseNetwork(network);
         setParseYearAired(year_aired);
         setParseUser(currUser);
+
+        // null check before adding to Parse
+        if(liked == null){
+            Log.i("Show Object", "No rating!");
+            liked = "";
+        }
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/models/Show.java:209]: Is this expected or handling an error case?  If it's an error, I would use `Log.e`

File: app/src/main/java/com/example/next_show/data/RecommendationClient.java (hunk 1 above)
Comment (alexisechano) [reply app/src/main/java/com/example/next_show/data/RecommendationClient.java:52]: If there are multiple saved shows, I don't want to do multiple Trakt API calls to get related shows back to back so in order to pick one, I just randomly choose one of the user's liked shows and grab data from there

File: app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java (hunk 4 above)
Comment (alexisechano) [reply app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java:111]: The Parse database doesn't accept strings being null so I have to initialize it as a blank string hence the check if the string actually equals a rating

File: app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java (hunk 5 above)
Comment (alexisechano) [reply app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java:130]: Yes! This is what I hope to fix this week and have the onClick trigger a saveInBackground for the database for the updated rating

**Pull request merged**